From 107b6e2d6139a8631839c855c0cfdcfc0a8d072e Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Sun, 1 Jan 2017 12:52:01 +0200
Subject: [PATCH 0704/1345] dts: a80x0: rename and cleanup of a8040-db device
 trees

commit  05a6d0aab98dc43fd7b7ff7b12acf59395a8c3d0 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Rename router setup to A0 setup
- Cleanup of unused DT
- Disabled all gop entries associated with disabled ppv2 units
- Removed all mdio phy entries associated with disabled gop units

Change-Id: I6dd6651805bd95965a063bb9b42df66c3ec885a3
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35206
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    2 +-
 .../mvebu/a7k-a8k/armada8040-db-setup.txt          |   10 +-
 arch/arm64/boot/dts/marvell/Makefile               |    2 +-
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   |   23 +-
 arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts  |  237 +++++++++++++++++++
 .../boot/dts/marvell/armada-8040-db-router.dts     |  242 --------------------
 6 files changed, 253 insertions(+), 263 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 4e722f4..127db82 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -35,7 +35,7 @@ Boards:
    - "marvell,armada8040-db-A", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
-   - "marvell,armada8040-db-router", "marvell,armada8040-db"
+   - "marvell,armada8040-db-A0", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
    - "marvell,armada8040-mcbin"
diff --git a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
index 7fb62c0..3b38cac 100644
--- a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
@@ -20,7 +20,7 @@ Board Setup
 ------------
 
 For A8040 A1 revision SoC - Please use armada-8040-db-A configuration
-For A8040 A0 revision SoC - Please use armada-8040-db-router configuration
+For A8040 A0 revision SoC - Please use armada-8040-db-A0 configuration
 
 ### setup A: armada-8040-db-A.dts  ###
 --------------------------------------
@@ -38,7 +38,7 @@ For A8040 A0 revision SoC - Please use armada-8040-db-router configuration
 			SW1, SW2, SW3, SW4, SW6, SW7, SW8 = OFF
 
 
-### setup 2: armada-8040-db-router.dts ###
+### setup A0: armada-8040-db-A0.dts ###
 ------------------------------------------
 
 	- Jumper-module (CON44): SLM1564
@@ -55,8 +55,8 @@ The tables below summarizes the interface configuration of each setup
 
 SERDES PHY configuration
 ------------------------
-Setup  #	|	A	|	B	| 	TBD	|
-Name   #	|     db-A	|    db-router	|		|
+Setup  #	|	A	|	A0	| 	TBD	|
+Name   #	|     db-A	|    db-A0	|		|
 -----------------------------------------------------------------
 CP0:								|
 -----------------------------------------------------------------
@@ -85,7 +85,7 @@ USB2_H0 (UTMI only)
 Multi-purpose pin configurations
 --------------------------------
 Setup  #	|	1	|	2	|	TBD	|
-Name   #	|     Default 	|    Router	|		|
+Name   #	|     Default 	|    A0		|		|
 -----------------------------------------------------------------
 AP806:								|
 -----------------------------------------------------------------
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 5a6eec9b..78eda30 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -18,7 +18,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-all-pcie.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-70x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU_PD) += armada-7040-pd.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A.dtb
-dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-router.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-A0.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-80x0-customer.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index f1ac384..0b971ec 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -42,13 +42,14 @@
 
 /*
  * Device Tree file for Marvell Armada 8040 development board
- * This file supports the default configuration of the board
+ * This file supports option A configuration
+ * CP0_RGMII1,CP1_RGMII0,CP0_EMMC,CP1_SPI(BOOT),CP1_UART,CP0_I2C,AP_eMMC,AP_UART
  */
 
 #include "armada-8040-db.dtsi"
 
 / {
-	model = "Marvell Armada-8040 development board router setup";
+	model = "Marvell Armada-8040 development board A setup";
 	compatible = "marvell,armada8040-db-A", "marvell,armada8040-db", "marvell,armada8040",
 			"marvell,armada-ap806-quad", "marvell,armada-ap806";
 	#address-cells = <2>;
@@ -77,7 +78,6 @@
 		config-space@f2000000 {
 			sata@540000 {
 				status = "okay";
-
 				sata-port@0 {
 					phys = <&comphy0 1 COMPHY_SATA0>;
 					phy-names = "comphy";
@@ -102,10 +102,7 @@
 			mdio@12a200 {
 				#address-cells = <1>;
 				#size-cells = <0>;
-				phy2: ethernet-phy@2 {
-					reg = <0>;
-				};
-				phy3: ethernet-phy@3 {
+				mdio0_phy1: ethernet-phy@1 {
 					reg = <1>;
 				};
 			};
@@ -116,12 +113,11 @@
 					phy-names = "comphy";
 				};
 				emac2: mac2 {
-					phy-mode = "rgmii"; /* rgmii-0 */
-					phy = <&phy2>;
+					phy-mode = "none"; /* disabled */
 				};
 				emac3: mac3 {
 					phy-mode = "rgmii"; /* rgmii-1 */
-					phy = <&phy3>;
+					phy = <&mdio0_phy1>;
 				};
 			};
 			ppv22@000000 {
@@ -160,7 +156,6 @@
 			};
 			sata@540000 {
 				status = "okay";
-
 				sata-port@0 {
 					phys = <&comphy1 1 COMPHY_SATA0>;
 					phy-names = "comphy";
@@ -175,7 +170,7 @@
 			mdio@12a200 {
 				#address-cells = <1>;
 				#size-cells = <0>;
-				phy2_1: ethernet-phy@2 {
+				mdio1_phy1: ethernet-phy@1 {
 					reg = <0>;
 				};
 			};
@@ -187,10 +182,10 @@
 				};
 				emac2_1: mac2 {
 					phy-mode = "rgmii"; /* rgmii-0 */
-					phy = <&phy2_1>;
+					phy = <&mdio1_phy1>;
 				};
 				emac3_1: mac3 {
-					phy-mode = "sgmii"; /* lane-4 */
+					phy-mode = "none"; /* disabled */
 				};
 			};
 			ppv22@000000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
new file mode 100644
index 0000000..4bb934e
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
@@ -0,0 +1,237 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 8040 development board
+ * This board file supports the A0 revision of the board
+ */
+
+#include "armada-8040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-8040 development board default A0 setup";
+	compatible = "marvell,armada8040-db-A0", "marvell,armada8040-db", "marvell,armada8040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			serial@512000 {
+				status = "okay";
+			};
+			sdhci@6e0000 {
+				status = "disabled";
+			};
+			i2c@511000 {
+				status = "okay";
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&ap_i2c0_pins>;
+				eeprom@57 {
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
+			};
+			spi0: spi@510600 {
+				status = "okay";
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			sata@540000 {
+				status = "okay";
+				sata-port@0 {
+					satus = "disabled";
+				};
+				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				mdio0_phy1: ethernet-phy@1 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "none"; /* disabled */
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&mdio0_phy1>;
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
+
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+	cpn-110-slave {
+		config-space {
+			usb3h0_1: usb3@500000 {
+				status = "okay";
+			};
+			sata@540000 {
+				status = "okay";
+				sata-port@0 {
+					satus = "disabled";
+				};
+				sata-port@1 {
+					phys = <&comphy1 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				mdio1_phy1: ethernet-phy@1 {
+					reg = <0>;
+				};
+			};
+			gop {
+				emac0_1: mac0 {
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "comphy";
+				};
+				emac2_1: mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&mdio1_phy1>;
+				};
+				emac3_1: mac3 {
+					phy-mode = "none"; /* disabled */
+				};
+			};
+			ppv22@000000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_ge1_rgmii_pins>;
+
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "okay";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index 0875114..e69de29 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -1,242 +0,0 @@
-/*
- * Copyright (C) 2016 Marvell Technology Group Ltd.
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPLv2 or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This library is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This library is distributed in the hope that it will be useful,
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively,
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use,
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-/*
- * Device Tree file for Marvell Armada 8040 development board
- * This board file supports the A0 revision of the board
- */
-
-#include "armada-8040-db.dtsi"
-
-/ {
-	model = "Marvell Armada-8040 development board router setup";
-	compatible = "marvell,armada8040-db-router", "marvell,armada8040-db", "marvell,armada8040",
-			"marvell,armada-ap806-quad", "marvell,armada-ap806";
-	#address-cells = <2>;
-	#size-cells = <2>;
-
-	chosen { };
-
-	ap806 {
-		config-space@f0000000 {
-			serial@512000 {
-				status = "okay";
-			};
-			sdhci@6e0000 {
-				status = "disabled";
-			};
-			i2c@511000 {
-				status = "okay";
-				clock-frequency = <100000>;
-				pinctrl-names = "default";
-				pinctrl-0 = <&ap_i2c0_pins>;
-				eeprom@57 {
-					compatible = "at24,24c64";
-					reg = <0x57>;
-					pagesize = <32>;
-				};
-				eeprom@50 {
-					compatible = "at24,24c64";
-					reg = <0x50>;
-					pagesize = <32>;
-				};
-			};
-			spi0: spi@510600 {
-				status = "okay";
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
-			};
-		};
-	};
-
-	cp110-master {
-		config-space@f2000000 {
-			sata@540000 {
-				status = "okay";
-
-				sata-port@0 {
-					satus = "disabled";
-				};
-				sata-port@1 {
-					phys = <&comphy0 3 COMPHY_SATA1>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-			};
-			usb3h0: usb3@500000 {
-				status = "okay";
-			};
-			usb3h1: usb3@510000 {
-				status = "okay";
-			};
-			sdhci@780000 {
-				status = "okay";
-			};
-			mdio@12a200 {
-				#address-cells = <1>;
-				#size-cells = <0>;
-				phy2: ethernet-phy@2 {
-					reg = <0>;
-				};
-				phy3: ethernet-phy@3 {
-					reg = <1>;
-				};
-			};
-			gop {
-				emac0: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&comphy0 2 COMPHY_SFI>;
-					phy-names = "comphy";
-				};
-				emac2: mac2 {
-					phy-mode = "rgmii"; /* rgmii-0 */
-					phy = <&phy2>;
-				};
-				emac3: mac3 {
-					phy-mode = "rgmii"; /* rgmii-1 */
-					phy = <&phy3>;
-				};
-			};
-			ppv22@000000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
-
-				eth0: eth0@010000 {
-					status = "okay";
-				};
-				eth1: eth1@020000 {
-					status = "disabled";
-				};
-				eth2: eth2@030000 {
-					status = "okay";
-				};
-			};
-			eip197@800000 {
-				status = "okay";
-			};
-			comphy {
-				status = "okay";
-			};
-		};
-
-		pcie@0x640000 {
-			status = "okay";
-		};
-	};
-	cp110-slave {
-		config-space@f4000000 {
-			usb3h0_1: usb3@500000 {
-				status = "okay";
-			};
-			sata@540000 {
-				status = "okay";
-
-				sata-port@0 {
-					satus = "disabled";
-				};
-				sata-port@1 {
-					phys = <&comphy1 3 COMPHY_SATA1>;
-					phy-names = "comphy";
-					status = "okay";
-				};
-			};
-			mdio@12a200 {
-				#address-cells = <1>;
-				#size-cells = <0>;
-				phy2_1: ethernet-phy@2 {
-					reg = <0>;
-				};
-			};
-			gop {
-				emac0_1: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&comphy1 2 COMPHY_SFI>;
-					phy-names = "comphy";
-				};
-				emac2_1: mac2 {
-					phy-mode = "rgmii"; /* rgmii-0 */
-					phy = <&phy2_1>;
-				};
-				emac3_1: mac3 {
-					phy-mode = "sgmii"; /* lane-4 */
-				};
-			};
-			ppv22@000000 {
-				pinctrl-names = "default";
-				pinctrl-0 = <&cps_ge1_rgmii_pins>;
-
-				eth0_1: eth0@010000 {
-					status = "okay";
-				};
-				eth1_1: eth1@020000 {
-					status = "okay";
-				};
-				eth2_1: eth2@030000 {
-					status = "disabled";
-				};
-			};
-			comphy {
-				status = "okay";
-			};
-		};
-		pcie@0x640000 {
-			status = "okay";
-		};
-	};
-};
-- 
1.7.9.5

